### DISCUSSION TITLE
Are multiple formatters for the same type supported in some way?

### DISCUSSION THREAD

**Discussion by Sacrilege** [Handlebars-Net/Handlebars.Net#424] [Q&A]
A example scenario would be a class that has 2 different DateTime properties where one should be rendered using the date and time and one should be rendered only using the date. Is there some way to tell handlebars to use a different formatter for each property?

**Comment by oformaniuk** [comment]
Hello @Sacrilege,

There's basically two options: 
- Explicitly define format in the template using helper
- Implement [IObjectDescriptorProvider](https://github.com/Handlebars-Net/Handlebars.Net/blob/master/source/Handlebars/ObjectDescriptors/IObjectDescriptorProvider.cs) and [IMemberAccessor](https://github.com/Handlebars-Net/Handlebars.Net/blob/master/source/Handlebars/MemberAccessors/IMemberAccessor.cs) for your type and return already formatted string for those properties.
An advanced variation of this would be to implement a generic wrapper that would know how to format containing value as well as accessing its properties.

Unfortunately defining custom format on property declaration (via attribute) is not supported and assuming current design it is complicated to implement something like this.
